src/libmongoc/src/mongoc/mongoc-cursor-cmd.c

```diff
@@ -216,5 +216,28 @@ _mongoc_cursor_cmd_new_from_reply (mongoc_client_t *client,
                       MONGOC_ERROR_CURSOR_INVALID_CURSOR,
                       "Couldn't parse cursor document");
    }
+
+   if (0 != cursor->cursor_id) {
```

Comment:
Logic could be simplified here being that there's no other work being done inside of this if statement. However, if you'd rather leave the if statements separate for readability and comments then that's fine as well.
```suggestion
      if (0 != cursor->cursor_id && 0 == cursor->server_id) {
```

Reply:
Seems like a reasonable simplification. Removed the nested `if`.